Civi/Api4/Service/Schema/SchemaMap.php

```diff
@@ -88,49 +88,19 @@ public function addTables(array $tables) {
   }
 
   /**
-   * Recursive function to traverse the schema looking for a path
+   * Traverse the schema looking for a path
    *
    * @param Table $table
    *   The current table to base fromm
    * @param string $target
    *   The target joinable table alias
-   * @param int $depth
-   *   The current level of recursion which reflects the number of joins needed
    * @param \Civi\Api4\Service\Schema\Joinable\Joinable[] $path
    *   (By-reference) The possible paths to the target table
-   * @param \Civi\Api4\Service\Schema\Joinable\Joinable[] $currentPath
-   *   For internal use only to track the path to reach the target table
    */
-  private function findPaths(Table $table, $target, $depth, &$path, $currentPath = []
-  ) {
-    static $visited = [];
-
-    // reset if new call
-    if ($depth === 1) {
-      $visited = [];
-    }
-
-    $canBeShorter = empty($path) || count($currentPath) + 1 < count($path);
-    $tooFar = $depth > self::MAX_JOIN_DEPTH;
```

Comment:
@colemanw this seems to be the only thing I wonder about with this, This seems to be something about trying to contort with something in MySQL. Is that replicated somewhere else?

Reply:
The long complicated code here was all made to handle a type of join which no longer exists in APIv4.
During alpha development, there was work done to support a pseudo-join where the api would run the main query, collect the ids and do a second query behind-the-scenes to gather e.g. contact.activities. This functionality has been dropped in favor of explicit joins.